index.html

```diff
@@ -2624,12 +2618,17 @@ <h2>WoT Runtime</h2>
             <p>
                 Technically, the <a>Thing</a> abstraction and its <a>Interaction Model</a> is implemented in a runtime system.
                 This <a>WoT Runtime</a> maintains the execution environment for the behavior implementation and is able to expose and/or consume <a>Things</a>,
-                and hence must be able to process, serialize, serve, and fetch <a>WoT Thing Descriptions</a>.
+                and hence must be able to fetch, process, serialize, and serve <a>WoT Thing Descriptions</a>.
             </p>
             <p>
-                The <a>WoT Runtime</a> must provide an API for the behavior implementation that follows the <a>Thing</a> abstraction.
-                The optional <a>WoT Scripting API</a> building block shown in <a href="#architecture-implementation"></a> defines such an application-facing interface.
+                The <a>WoT Runtime</a> must provide an API for the behavior implementation.
+                The optional <a>WoT Scripting API</a> building block shown in <a href="#architecture-implementation"></a>
+                defines such an application-facing interface that follows the <a>Thing</a> abstraction.
                 See <a href="#native-impl"></a> for alternative APIs.
+                In general, application logic should be executed in isolated <a>Execution Environments</a>
```

Comment:
An Execution Environment is not necessarily isolated. Then, you should specify what level of isolation do you mean.